src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java

```diff
@@ -49,36 +49,34 @@ public FtpClient(
     }
     // endregion
 
-    public void upload(FileToSend file, String serviceFolder) {
-        Instant start = Instant.now();
+    public void upload(FileToSend file, String serviceFolder, SFTPClient sftpClient) {
+        String folder = file.isSmokeTest
+            ? configProperties.getSmokeTestTargetFolder()
+            : String.join("/", configProperties.getTargetFolder(), serviceFolder);
 
-        runWith(sftp -> {
-            boolean isSuccess = false;
-
-            try {
-                String folder = file.isSmokeTest
-                    ? configProperties.getSmokeTestTargetFolder()
-                    : String.join("/", configProperties.getTargetFolder(), serviceFolder);
+        String path = String.join("/", folder, file.getName());
+        Instant start = Instant.now();
+        boolean isSuccess = false;
 
-                String path = String.join("/", folder, file.getName());
-                sftp.getFileTransfer().upload(file, path);
+        try {
+            sftpClient.getFileTransfer().upload(file, path);
 
-                logger.info(
-                    "File uploaded. Time: {}, Size: {}, Folder: {}",
-                    ChronoUnit.MILLIS.between(start, Instant.now()) + "ms",
-                    file.content.length / 1024 + "KB",
-                    serviceFolder
-                );
+            isSuccess = true;
+        } catch (IOException exc) {
+            throw new FtpException("Unable to upload file.", exc);
+        } finally {
+            long millis = ChronoUnit.MILLIS.between(start, Instant.now());
 
-                isSuccess = true;
+            insights.trackFtpUpload(Duration.ofMillis(millis), isSuccess);
 
-                return null;
-            } catch (IOException exc) {
-                throw new FtpException("Unable to upload file.", exc);
-            } finally {
-                insights.trackFtpUpload(Duration.between(start, Instant.now()), isSuccess);
-            }
-        });
+            logger.info(
```

Comment:
I can see it was already like this before this PR, so I'm fine with it staying as it is, but it would be worth loggin errors in case of failures at some point here.

Reply:
Having another look, I think this is the only thing we'll log if we fail to upload a file. As a result, we'll have no errors logged. Doesn't have to be addressed in this PR and it's nothing you've done, but needs to be handled at some point.

Reply:
This used to be logged in `try` block, move back. We don't want this in case of failures.

Reply:
Failure logging is automatically done by app insights by pushing exception telemetry which has not been caught aka failure case is logged. This log entry is informational analog of dependency tracking but with finer details in it. Slightly amended the message to include yes/no and full path where it was attempted to go instead of just folder. Exception message includes full path too so it can be tracked back to this informational material

Reply:
The info message says `File uploaded: ...`, which may actually mean `Failed to upload file`. While there is a flag later in the message that says whether it's a success of failure, it's going to be very easy to misinterpret it. I'm a huge fan of info saying `File uploaded` and error saying `Error when uploading file` - no way to get those messages wrong.

Reply:
I am fine with both
1. Either log separately like @lgonczar suggested or
2. Update log message to say File uploaded successfully ? Yes ...

Reply:
Keep the changes around single connection, please.
Feel free to introduce other changes in separate PR(s).